tidepods/cli.py

```diff
@@ -1,50 +1,97 @@
 import click
 from datetime import datetime as dt
 
+
 def click_callback(f):
     return lambda _, __, x: f(x)
-    
+
+
 def check_date(date):
     try:
-        return dt.strptime(date , "%Y%m%d")
+        return dt.strptime(date, "%Y%m%d")
     except ValueError:
         msg = "Not a valid date: '{0}'.".format(date)
         raise Exception(msg)
-        
+
+
+def check_time(time):
+    try:
+        return dt.strptime(time, "%H:%M")
+    except ValueError:
+        msg = "Not a valid timestamp: '{0}'.".format(time)
+        raise Exception(msg)
```

Comment:
Never raise `Exception`. Be more descriptive. This is indeed a `ValueError`, since an invalid value has been passed.

The Python 3 way, which conserves the original traceback, is

```python
try:
    ...
except ValueError as exception:
    raise ValueError(msg) from exception
```

Reply:
Done